File: CollatzProblemi/CollatzProblemi.java
package collatzproblemi;

import java.util.Scanner;

public class CollatzProblemi {

    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        System.out.print("Bir sayı girin : ");
        int sayi = reader.nextInt();
        while (sayi > 1) {
            System.out.print(sayi + " ");
            if (sayi % 2 == 0) {
                sayi /= 2;
            } else {
                sayi = 3 * sayi + 1;
            }
        }
        System.out.println("1");
    }
}

File: CollatzProblemi/CollatzProblemi.py
def collatz_problemi(ust_sayi):

    while ust_sayi > 1:
        print(int(ust_sayi), end=' ')
        if ust_sayi % 2 == 0:
            ust_sayi /= 2
        else:
            ust_sayi = 3 * ust_sayi + 1
    else:
        print(1)


sayi = int(input("Bir sayı girin : "))
collatz_problemi(sayi)